src/Controllers/WebChat/ConversationController.php

```diff
@@ -62,19 +64,22 @@ public function handleNewConversation(Map $utterance)
             $actionResult = $this->getAgent()->performAction($action, $utterance);
         }
 
+        $informationResponse = null;
+        if ($informationRequest = $ci->getCurrentInformationRequest()) {
+            $informationResponse = $this->getAgent()->performInformationRequest($informationRequest, $utterance);
+        }
+
         /* @var \actsmart\actsmart\Conversations\Utterance $nextUtterance */
         $nextUtterance = $ci->getNextUtterance($this->getAgent(), $utterance, $intent, false);
 
         $this->getAgent()->getActuator('actuator.webchat')->perform('action.webchat.postmessage', [
-            Literals::MESSAGE => $nextUtterance->getMessage()->getWebChatResponse($actionResult ?? $utterance)
+            Literals::MESSAGE => $nextUtterance->getMessage()->getWebChatResponse($actionResult ?? $utterance, $informationResponse)
```

Comment:
Thinking we need to structure the getWebChatResponse parameter more (potentially an object from which you can extract an action result, or an information response or an utterance). The message "template" can then look for the information it needs  and deal with the cases where it doesn't find it.

Reply:
Yes, I agree. And I think that this is something that can be standardised across all message types (Slack, Facebook etc)